File: 11. Container With Most Water/container_with_most_water.php
<?php
function absolute ($number){
	return (($number < 0) ? ($number * -1) : ($number * 1));
}

function getVolume ($height1, $height2, $width){
	$height = ($height1 < $height2) ? ($height1) : ($height2);
	
	return ($height * $width);
}

function containerPoints ($list){
	$lenList = count($list);
	$largestVolume = 0;
	$volumeIndixes = [];
	
	for ($a = 0; $a < $lenList; $a++){
		for ($b = ($lenList - 1); $b > 0; $b--){
			if ($a == $b){
				continue;
			} else if ($b < $a){
				break;
			} else {
				$interval = absolute ($a - $b);
				$volume = getVolume ($list[$a], $list[$b], $interval);
				
				if ($largestVolume < $volume){
					$largestVolume = $volume;
					$volumeIndexes = [$a, $b];
				}
			}
		}
	}
	
	return ($volumeIndexes);
}

$heightList = [1,8,6,2,5,4,8,3,7];
$heightIndices = containerPoints($heightList);
$height1 = $heightList[$heightIndices[0]];
$height2 = $heightList[$heightIndices[1]];
$interval = absolute($heightIndices[0] - $heightIndices[1]);

echo("This is the handy work of PHP\n\n");
echo(json_encode($heightIndices) . "\n");
echo("The heights given the points are: " . strval($height1) . " on one end and " . strval($height2) . " on the other end. The difference between the two points is " . strval($interval) . " giving a total volume of " . strval(getVolume($height1, $height2, $interval)) . "\n\n");
?>

File: 11. Container With Most Water/container_with_most_water.rs
struct HeightPoint{
	heightPoint1: isize,
	heightPoint2: isize,
}

fn absolute (number: isize) -> isize{
	return (if (number < 0)  {(number * -1)} else {(number * 1)});
}

fn get_volume (height1: isize, height2: isize, width: isize) -> isize{
	let height = if (height1 < height2) {(height1)} else {(height2)};
	
	return (height * width);
}

fn container_points(list: &[isize]) -> HeightPoint{
	let len_list: isize = list.len() as isize;
	let mut out: HeightPoint = HeightPoint{heightPoint1: 0, heightPoint2: 0};
	let mut largest_volume: isize = 0;
	
	for a in 0..len_list{
		for b in (0..(len_list)).rev(){
			if a == b{
				continue;
			} else if b < a{
				break;
			} else {
				let interval: isize = absolute((a - b) as isize);
				let volume: isize = get_volume (list[a as usize], list[b as usize], interval);
				//println!("{}\t{}\t{}\t{}\t{}\t{}", a, b, list[a as usize], list[b as usize], interval, volume);
				
				if largest_volume < volume{
					largest_volume = volume;
					
					out.heightPoint1 = a;
					out.heightPoint2 = b;
				}
			}
		}
	}
	return out;
}

fn main (){
	let height_list: &[isize] = &[1,8,6,2,5,4,8,3,7];
	let len_height_list: isize = height_list.len() as isize;
	
	let height_indices: HeightPoint = container_points(height_list);
	
	let height1: isize = height_list[height_indices.heightPoint1 as usize];
	let height2: isize = height_list[height_indices.heightPoint2 as usize];
	
	let interval: isize = absolute(height_indices.heightPoint1 - height_indices.heightPoint2);
	
	println!("This is the handy work of Rust\n\n");
	
	println!("[{}, {}]", height_indices.heightPoint1, height_indices.heightPoint2);
	
	println!("The heights given the points are: {} on one end and {} on the other end. The difference between the two points is {} giving a total volume of {}\n\n\n", height1, height2, interval, get_volume(height1, height2, interval));
}
